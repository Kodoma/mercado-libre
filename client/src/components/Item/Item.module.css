.root {
    margin: 0 10%;
    display: flex;
    flex-direction: column;
    min-width: fit-content;
    width: 500px;
    background-color: #EEEEEE;
}

.cover {
    cursor: zoom-in;
    height: 480px;
    object-fit: contain;
    width: 380px;
}
.details {
    display: flex;
    flex-direction: column;
}
.content {
    flex: 1 0 auto;
}

.customTag {
    max-width: 100%;
    height: 500px;
    background: black;
}

.imageGallery {
    width: 25%;
}

.data {
    padding-left: 10%;
    padding-top: 5%;
    color: #333333;
}
.sold {
    color: #707070;
}

@media (max-width: 812px) {
    .root {
        width: 100%;
        min-width: auto;
    }
    .data {
        padding: 0px 0px 15% 0px;
        margin-left: auto !important;
        margin-right: auto !important;
    }
}
