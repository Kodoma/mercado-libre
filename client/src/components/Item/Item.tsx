import React from 'react';
import { Item } from '../../models/items';
import fetch from 'cross-fetch';
import styles from './Item.module.css';
import { Button, Card, CardContent, Grid, Typography } from '@material-ui/core';
import Breadcrumb from '../BreadCrumb/Breadcrumb';
import ImageGallery from './ImageGallery/ImageGallery';
import Price from '../Items/Price/Price';
import { isMobile } from 'react-device-detect';
import { useRouter } from 'next/router';
import Loading from '../Loading/Loading';

const ItemProduct = () => {
    const [item, setItem] = React.useState(ItemProductInitState);
    const router = useRouter();
    const { id } = router && router.query;

    React.useEffect(() => {
        async function fetchItemProduct(id: string) {
            if (!id) return;
            const request = await fetch('http://0.0.0.0:8080/api/items/' + id);
            if (request.status === 200) {
                const response = await request.json();
                setItem(response.item);
            } else {
                await router.push('/error');
            }
        }

        fetchItemProduct(id as string);
    }, [id]);

    const content = (
        <div className={styles.root}>
            <Breadcrumb search={item && item.title} />
            <Card>
                <CardContent>
                    <Grid
                        container
                        alignItems={isMobile ? 'center' : 'flex-start'}
                        direction={isMobile ? 'column' : 'row'}>
                        <Grid item>
                            <ImageGallery pictures={item.pictures} />
                        </Grid>
                        <Grid item xs={isMobile ? 1 : 6} className={styles.data}>
                            <Grid item xs>
                                <Grid item xs>
                                    <Typography variant="subtitle1" className={styles.sold}>
                                        {`${item.condition} - ${item.sold_quantity} vendidos`}
                                    </Typography>
                                    <Typography variant="h6" gutterBottom>
                                        {item.title}
                                    </Typography>
                                    <Price
                                        id={item.id}
                                        amount={item.price.amount}
                                        currency={item.price.currency}
                                        free_shipping={item.free_shipping}
                                    />
                                    <Button
                                        variant="contained"
                                        color="primary"
                                        fullWidth={isMobile ? true : false}>
                                        Comprar
                                    </Button>
                                </Grid>
                            </Grid>
                        </Grid>
                        {item.description.length > 0 && (
                            <Grid container direction="row">
                                <Typography variant="h6" gutterBottom>
                                    Descripcion del producto.
                                </Typography>
                                <Typography variant="body2" color="textSecondary">
                                    {item.description}
                                </Typography>
                            </Grid>
                        )}
                    </Grid>
                </CardContent>
            </Card>
        </div>
    );

    return <React.Fragment>{item === ItemProductInitState ? <Loading /> : content}</React.Fragment>;
};

export default React.memo(ItemProduct);

const ItemProductInitState: Item = {
    id: '',
    title: '',
    price: {
        amount: '',
        currency: ''
    },
    thumbnail: '',
    pictures: [],
    condition: '',
    free_shipping: false,
    sold_quantity: 0,
    description: ''
};
