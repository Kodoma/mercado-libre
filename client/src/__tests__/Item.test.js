import React from 'react';
import 'jsdom-global/register';
import {shallow} from 'enzyme';
import Item from "../components/Item/Item";

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('tests Search', () => {
    it('should render my Item component', () => {
        useRouter.mockImplementationOnce(() => ({
            query: { id: 'MLA901420435' },
        }))

        const component = shallow(<Item />);
        expect(component.getElements()).toMatchSnapshot();
    });

});