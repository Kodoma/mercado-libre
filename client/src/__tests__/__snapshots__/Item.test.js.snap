// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests Search should render my Item component 1`] = `
Array [
  <React.Fragment>
    <Loading />
  </React.Fragment>,
]
`;
